File: Python/strStr.py

haystack = "mississippi"
needle = "sipp"

def strStr(haystack, needle):
    

    for x in range(len(haystack) - len(needle) + 1):
        if haystack[x:x + len(needle)] == needle: 
            return x
                
    return -1

print(strStr(haystack, needle))

File: JavaScript/strStr.js


let haystack = "mississippi"
let needle = "sipp"

var strStr = function(haystack, needle) {
    for (let x = 0; x <= haystack.length - needle.length; x++) {

        if (haystack.substring(x, x + needle.length) === needle) {
           return x
        }
    }  
    return -1
};

console.log(strStr( haystack,needle))